services/QuillLMS/app/controllers/auth/google_controller.rb

```diff
@@ -1,28 +1,43 @@
 class Auth::GoogleController < ApplicationController
-  before_action :set_profile,                     only: :google
-  before_action :set_user,                        only: :google
-  before_action :save_teacher_from_google_signup, only: :google
-  before_action :save_student_from_google_signup, only: :google
-  before_action :follow_google_redirect,          only: :google
-
-  def google
-    if @user.teacher?
-      GoogleStudentImporterWorker.perform_async(@user.id, 'Auth::GoogleController')
-    end
+  before_action :set_profile, only: [:authorization_and_authentication, :authentication]
```

Comment:
before_action can [take an array of callbacks](https://api.rubyonrails.org/v6.1.4/classes/AbstractController/Callbacks/ClassMethods.html#method-i-before_action). Let's use this style here, to increase readability / reduce LoC

Reply:
Good idea